src/machine/machine_stm32.go

```diff
@@ -75,3 +96,69 @@ func (p GPIO) Set(high bool) {
 		port.BSRR = 1 << (pin + 16)
 	}
 }
+
+// UART
+var (
+	// USART1 is the first hardware serial port on the STM32.
+	// Both UART0 and UART1 refers to USART1.
+	UART0 = &UART{}
+	UART1 = UART0
```

Comment:
So there is no UART0 on the STM32?
It seems a bit wrong this way but I guess exposing UART0 for USART1 is the best thing to do here, then.

Reply:
Somewhat offtopic: I think it would be nice to not have `UART0`, `UART1 etc. but just `[...]UART` at some point in the future, to also make it easier to enumerate UART devices, for example in periph. This is even more relevant to other peripherals like I2C, especially on chips that don't have fixed pins for peripherals.